app.go

```diff
@@ -268,7 +268,7 @@ func (a *App) RunContext(ctx context.Context, arguments []string) (err error) {
 		return err
 	}
 
-	if !a.HideHelp && checkHelp(context) {
+	if !a.HideHelp && (checkHelp(context) || checkHelpArguments(arguments)) {
```

Comment:
The `arguments` variable here I believe is the unparsed arguments, rather than the parsed arguments.

What that means from a practical perspective is that if we were to have flags like this:

```
my-app --string-flag foo --int-flag 15 text1 text2
```

The unparsed args would be: `["my-app", "--string-flag", "foo", "--int-flag", "15", "text1", "text2"]`

The parsed args would be: `["text1", "text2"]`

The reason this is important is because we want to distinguish between when `help` is passed as the value of a flag (e.g., `--string-flag help` as opposed to as a command. 

I believe the way to do that would be to access `context.Args()`,  which under the hood uses a `flag.FlagSet` that has already parsed our flags. This also has the added benefit of letting us modify the definition of `checkHelp`, as opposed to creating a new function that must be called alongside it. But we'll definitely want to rely on good test cases to prove that it works as expected 🙂

Reply:
ah! this is the kind of pointer I needed! I'll rework this part a bit. I just ran into a bit of trouble with `TestShowCommandHelp_HelpPrinter` showing the wrong template. I think we just need to know if there is a command being called or not. I'll dig a bit further soon 😄